src/layouts/Roadmap.tsx

```diff
@@ -53,7 +53,9 @@ export const roadmapComponents = {
   RoadmapImageContent,
 }
 
-interface IProps extends MdPageContent, ChildOnlyProp {
+interface IProps
+  extends ChildOnlyProp,
+    Pick<MdPageContent, "slug" | "tocItems"> {
```

Comment:
Constraining the prop type for all the layout components based on the props being used.

Reply:
I think I understand the intention here but it is not truth.
```ts
// we still receive these props in each layout
const layoutProps = {
  slug,
  frontmatter,
  lastUpdatedDate,
  timeToRead,
  tocItems,
}
```
so, I think we are kind of lying to ourselves here, props being used vs being passed.

Not sure what would be the best way to go. I'm happy to keep it like this for now.

Sidenote: might be a good idea to accept a param in `MdPageContent<RoadmapFrontmatter>` for the frontmatter type.

Reply:
@pettinarip
Since that object was already present, I was only creating constraints with this object in mind! 😅 

I'm not sure what to do about either though. If the new DS is looking to scale down the number of layouts, then this may become a non-issue and would be fine to leave this for now. Otherwise the only thing I can think of is keeping some of the logic in the layout component and not be passed from `getStaticProps`.

> Sidenote: might be a good idea to accept a param in MdPageContent<RoadmapFrontmatter> for the frontmatter type.

Maybe this:
```ts
/**
  * This generic declaration assigns the `Frontmatter` intersection type without extending it.
  *
  * This both makes the generic optional in re-declaring, and prevent errors in missing types when passing
  * a frontmatter type specific to the layout.
  */
export interface MdPageContent<F = Frontmatter> {
  slug: string
  content: string
  frontmatter: F
  tocItems: Array<ToCItem>
  lastUpdatedDate?: string
  contentNotTranslated: boolean
}

// In context
type DocsLayoutProps = ChildOnlyProp & MdPageContent<DocsFrontmatter>
```

Reply:
All good, lets see how it goes.

re: frontmatter type, yes, that looks good.